enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java

```diff
@@ -766,4 +778,17 @@ private String getJWTTokenIdentifier(SignedJWTInfo signedJWTInfo) {
         }
         return signedJWTInfo.getSignedJWT().getSignature().toString();
     }
+
+    public String extractJWTInWSProtocolHeader(RequestContext requestContext) {
+        String protocolHeader = requestContext.getHeaders().get(
+                HttpConstants.WEBSOCKET_PROTOCOL_HEADER);
+        if (protocolHeader != null) {
+            String[] secProtocolHeaderValues = protocolHeader.split(",");
+            if (secProtocolHeaderValues.length > 1 && secProtocolHeaderValues[0].equals(
```

Comment:
What if WEBSOCKET_PROTOCOL_HEADER contains `chat, choreo-token, <TOKEN>`?

Reply:
The user is supposed to send in the given order. First token information and then backend protocol information. 

We have implemented in a similar way for the handling of API key and the console sends the information in this format.